File: 217-Contains-Duplicate.py
class Solution(object):
	def containsDuplicate(self, nums):
		unique_nums = set()

		for i in range(len(nums)):
			if nums[i] in unique_nums:
				return True
			unique_nums.add(nums[i])
		return False

	def containsDuplicate(self, nums):
		return len(set(nums)) != len(nums)

File: 217-Contains-Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> set;

        for (int num : nums) {
            if (set.find(num) != set.end())
                return true;
                
            set.insert(num);
        }

        return false;
    }
};